src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx

```diff
@@ -96,13 +97,35 @@ function LineNumberTooltip({ editor, keyModifiers }: Props) {
     var rect = editor.codeMirror.getWrapperElement().getBoundingClientRect();
     var topVisibleLine = editor.codeMirror.lineAtHeight(rect.top, "window");
     var bottomVisibleLine = editor.codeMirror.lineAtHeight(rect.bottom, "window");
-    dispatch(fetchHitCounts(source!.id, editor.codeMirror.lineAtHeight()));
-    dispatch(fetchHitCounts(source!.id, topVisibleLine - 10));
-    dispatch(fetchHitCounts(source!.id, bottomVisibleLine + 10));
+
+    const viewport = editor.editor.getViewport();
+    const { from: topViewportLine, to: bottomViewportLine } = viewport;
+    const centerLine = ((bottomViewportLine - topViewportLine) / 2) | 0;
+
+    // We want to try to ensure we have hit counts above and below the viewport
+    // Can't go less than line index 0, though
+    const bufferAboveLine = Math.max(topVisibleLine - 10, 0);
+    const bufferBelowLine = bottomVisibleLine + 10;
+
+    // But, some of these lines could belong to the same 100-line chunk
+    const lineBounds = [centerLine, bufferAboveLine, bufferBelowLine].map(line =>
+      getBoundsForLineNumber(line)
+    );
+    const lineBoundsMap = new Map<number, { lower: number; upper: number }>();
+
+    // Deduplicate bounds chunks based on their `lower` line number
+    for (let entry of lineBounds) {
+      lineBoundsMap.set(entry.lower, entry);
+    }
+
+    // Now try to fetch hit counts for the unique bounds chunks
+    for (let bounds of lineBoundsMap.values()) {
```

Comment:
This feels like a lot of ceremony to `uniq` these values. Would it be clearer to just call `uniq`?

Reply:
Hmm. Yeah, I can maybe try `uniqBy`?